### DISCUSSION TITLE
Optimising a 3 dimensional shape

### DISCUSSION THREAD

**Discussion by olkovi** [mitsuba-renderer/mitsuba3#795] [Q&A]
So, in this [tutorial](https://www.youtube.com/watch?v=eTHL3W2NUn0) it appears that it is possible to one surface of a plate to produce a desired light distribution. I wonder if a more complex 3D shape could be achieved similarly?

**Comment by njroussel** [comment]
Hi @olkovi 

If I understand your question correctly, yes it is possible. The surface being optimized is just a triangle mesh that happens to be mostly flat, but in fact it could be a sphere or something similar.
To be clear, I'm just saying that it is technically possible - from an implementation standpoint. I do not know how it would actually behave in practice: too many local minimas? too ambiguous? etc.
